sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -90,4 +92,10 @@ class DataFrameFunctionsSuite extends QueryTest {
       testData2.select(bitwiseNOT($"a")),
       testData2.collect().toSeq.map(r => Row(~r.getInt(0))))
   }
+
+  test("bin") {
+    checkAnswer(
```

Comment:
can we use a locally scoped dataset and explicitly put down the result? for example
```
val df = Seq[(java.lang.Integer, java.lang.Integer)]((123, null)).toDF("a", "b")
checkAnswer(
  df.select(bin("a"), bin($"b")),
  Row("1100", null))
```